packages/astro/src/runtime/server/endpoint.ts

```diff
@@ -42,6 +42,27 @@ export async function renderEndpoint(
 	const response = await handler.call(mod, context);
 	// Endpoints explicitly returning 404 or 500 response status should
 	// NOT be subject to rerouting to 404.astro or 500.astro.
-	response.headers.set(REROUTE_DIRECTIVE_HEADER, 'no');
-	return response;
+	return setRerouteDirective(response);
+}
+
+// This is separated into a dedicated function,
+// to allow the JavaScript runtime to optimize
+// the `renderEndpoint` implementation.
+// For more details: https://web.dev/articles/speed-v8#therefore_5
+function setRerouteDirective(response: Response): Response {
+	try {
+		response.headers.set(REROUTE_DIRECTIVE_HEADER, 'no');
+
+		return response;
+	} catch (_) {
```

Comment:
Do you think the catch block could removed altogether?

It only fails with a 300 response, which wouldn't have rerouted anyways.

Now that I think about it, we could only set the header when response status is 404 or 500 which might be even less code.

Reply:
Yes, I think you are right, in most cases it is unnecessary to set the reroute directive header.

Should i change it to only add the header if the HTTP status code is 404 or 500?